From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect l2tp_tunnel_delete
Patch-mainline: never, kabi
References: kabi

In 4.4.94, commit b5f689d94bc3bbcaf25142f43bd31712d10adb44 (l2tp: fix
race condition in l2tp_tunnel_delete), upstream commit
62b982eeb4589b2e6d7c01a90590e3a4c2b2ca19 changed return type of
l2tp_tunnel_delete from int to void. This indeed changed the checksum of
this exported function and the kABI checker now complains.

So revert this change, making l2tp_tunnel_delete to return int again.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 net/l2tp/l2tp_core.c |    4 +++-
 net/l2tp/l2tp_core.h |    2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

--- a/net/l2tp/l2tp_core.c
+++ b/net/l2tp/l2tp_core.c
@@ -1686,12 +1686,14 @@ EXPORT_SYMBOL_GPL(l2tp_tunnel_create);
 
 /* This function is used by the netlink TUNNEL_DELETE command.
  */
-void l2tp_tunnel_delete(struct l2tp_tunnel *tunnel)
+int l2tp_tunnel_delete(struct l2tp_tunnel *tunnel)
 {
 	if (!test_and_set_bit(0, &tunnel->dead)) {
 		l2tp_tunnel_inc_refcount(tunnel);
 		queue_work(l2tp_wq, &tunnel->del_work);
+		return 1;
 	}
+	return 0;
 }
 EXPORT_SYMBOL_GPL(l2tp_tunnel_delete);
 
--- a/net/l2tp/l2tp_core.h
+++ b/net/l2tp/l2tp_core.h
@@ -260,7 +260,7 @@ int l2tp_tunnel_create(struct net *net,
 		       u32 peer_tunnel_id, struct l2tp_tunnel_cfg *cfg,
 		       struct l2tp_tunnel **tunnelp);
 void l2tp_tunnel_closeall(struct l2tp_tunnel *tunnel);
-void l2tp_tunnel_delete(struct l2tp_tunnel *tunnel);
+int l2tp_tunnel_delete(struct l2tp_tunnel *tunnel);
 struct l2tp_session *l2tp_session_create(int priv_size,
 					 struct l2tp_tunnel *tunnel,
 					 u32 session_id, u32 peer_session_id,
